### PR TITLE
feat(TFD-1592): API changes for components part 1

### PR DESCRIPTION
**Pull request by bchen-talend**
**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**
BigQuery/Pubsub API changes except PipelineOptions change
Elasticsearch/Kafka/FixedFlowInput/FilterRow API changes, IT pass

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: components/components-bigquery/bigquery-runtime/src/main/java/org/talend/components/bigquery/runtime/BigQueryInputRuntime.java [hunk 1]
```diff
@@ -72,28 +69,28 @@ public ValidationResult initialize(RuntimeContainer container, BigQueryInputProp
...
-            // when using Dataflow runner, these properties has been set on pipeline level
-            gcpOptions.setProject(datastore.projectName.getValue());
-            gcpOptions.setTempLocation(datastore.tempGsFolder.getValue());
-            gcpOptions.setCredentialFactoryClass(ServiceAccountCredentialFactory.class);
-            gcpOptions.setServiceAccountFile(datastore.serviceAccountFile.getValue());
-            gcpOptions.setGcpCredential(BigQueryConnection.createCredentials(datastore));
-        }
-
-        BigQueryIO.Read.Bound bigQueryIOPTransform;
+//        GcpServiceAccountOptions gcpOptions = in.getPipeline().getOptions().as(GcpServiceAccountOptions.class);
```
Comment (bchen-talend) [inline components/components-bigquery/bigquery-runtime/src/main/java/org/talend/components/bigquery/runtime/BigQueryInputRuntime.java]: this part will be supported in next PR

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/772/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/772/)

File: components/components-bigquery/bigquery-runtime/src/main/java/org/talend/components/bigquery/runtime/BigQueryDatasetRuntime.java [hunk 2]
```diff
@@ -71,7 +70,6 @@ public void getSample(int limit, Consumer<IndexedRecord> consumer) {
         // Create a pipeline using the input component to get records.
         DirectOptions options = BeamLocalRunnerOption.getOptions();
         final Pipeline p = Pipeline.create(options);
-        LazyAvroCoder.registerAsFallback(p);
```
Comment (pbailly) [inline components/components-bigquery/bigquery-runtime/src/main/java/org/talend/components/bigquery/runtime/BigQueryDatasetRuntime.java]: Why are you removing the initialization step of the LazyAvroCoder?

Comment (bchen-talend) [inline components/components-bigquery/bigquery-runtime/src/main/java/org/talend/components/bigquery/runtime/BigQueryDatasetRuntime.java]: Beam change to use AutoService ranther than registerFallback for coder registry, so we change to use [LazyAvroCoderProviderRegistrar.java](https://github.com/Talend/components/blob/bchen/TFD-1396/Beam2-Integration/core/components-adapter-beam/src/main/java/org/talend/components/adapter/beam/coders/LazyAvroCoderProviderRegistrar.java)

**Pull request merged**